bin/vimeo-linker.py

```diff
@@ -0,0 +1,216 @@
+    res = v.get(requestUrl)
+    if res == None:
+        print("Result was None; sleeping and trying again")
+        time.sleep(2)
+        continue
+    numRequests += 1
+    j = res.json()
+    if not res.ok:
+        print("could not fetch videos from vimeo API!")
+        exit(1)
+    for elem in j["data"]:
+        nameUrls[elem["name"]] = elem["link"]
+    requestUrl = j["paging"]["next"]
+    print(requestUrl)
+    cont = requestUrl != None
+    # seems to be needed to not run into read timeouts.
+    time.sleep(1)
+
+if fetchUrls and cacheVimeo:
+    pickle.dump(nameUrls, open("videos.pickle", "wb"))
+    
+
+notFounds = []
+result = ""
+matcher = difflib.SequenceMatcher()
+
+
+def trySubstringMatch(videoName):
+    for idx, p in papers.items():
+        if p.is_volume:
+            continue
+        title = p.get_title("plain")
+        if len(title) <=  20:
+            continue
+        if title.lower() in videoName:
+            return (idx, p)
+    return None
+    
+def tryMatch(name):
+    matcher.set_seq1(name)
+    for idx, p in papers.items():
+        if p.is_volume:
+            continue
+        title = p.get_title("plain")
+        matcher.set_seq2(title.lower())
+        if matcher.real_quick_ratio() > 0.8 and \
+         matcher.quick_ratio() > 0.8 and \
+         matcher.ratio() > 0.8:
+            return (idx, p)
+    return None
+
+num_elems = len(nameUrls)
+
+
+id_video = []
+
+computeMatch = True
+if os.path.isfile("videos_papers.pickle") and cacheMatchings:
+    id_video = pickle.load(open("videos_papers.pickle", "rb"))
+    computeMatch = False
+
+
+if computeMatch:
+    i = 1
+    for name, url in nameUrls.items():
+        print(i, " of ", num_elems, end="")
+        i += 1
+        # clean tacl tag and author list
+        if name.startswith("[TACL]"):
+            name = name[6:]
+        if "---" in name:
+            name = name.split("---")[0]
+        found = False
+        # skip video names that are obviously not paper titles
+        # because they don't contain spaces.
+        if " " in name:
+            name = name.lower()
+            res = trySubstringMatch(name)
+            if res == None:
+                res = tryMatch(name)
+            # try to remove author list appended by : or by -
+            if res == None:
+                if ":" in name:
+                    res = tryMatch(name.rsplit(":", 1)[0])
+            if res == None:
+                if "-" in name:
+                    res = tryMatch(name.rsplit("-", 1)[0])
+
+            if res != None:
+                (idx, p) = res
+                if checkVideo(p):
+                    print("video already exists, skipping ...")
+                    continue
+                title = p.get_title("plain")
+                print("found title " + title + " for video " + name)
+                result += title +"\t"+ idx +"\t"+ url +"\t"+ name +"\t"+ p.get_booktitle("plain") + "\n"
+                id_video.append((idx, url))
```

Comment:
An f-string here (since we require Python 3.7 anyway) would make this more readable.